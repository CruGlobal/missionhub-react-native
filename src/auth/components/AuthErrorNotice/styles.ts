import { StyleSheet } from 'react-native';

import theme from '../../../theme';

export const styles = StyleSheet.create({
  errorBar: {
    backgroundColor: '#FF5532',
    height: 48,
    justifyContent: 'center',
    alignItems: 'center',
  },
  errorMessage: {
    color: theme.white,
    fontSize: 16,
  },
});
