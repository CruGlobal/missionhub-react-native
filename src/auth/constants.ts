import { AuthPerson_currentUser_person } from './__generated__/AuthPerson';

export enum IdentityProvider {
  Apple,
  Google,
  Facebook,
  TheKey,
  Anonymous,
}

export enum AuthError {
  None,
  CredentialsIncorrect,
  EmailUnverified,
  MfaRequired,
  MfaIncorrect,
  Unknown,
}

export const emptyAuthPerson: AuthPerson_currentUser_person = {
  __typename: 'Person' as const,
  id: '',
  firstName: '',
  lastName: '',
  fullName: '',
  stage: null,
  globalRegistryMdmId: null,
  fbUid: null,
  theKeyUid: null,
};
