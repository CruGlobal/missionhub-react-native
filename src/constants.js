export const LOGIN = 'app/LOGIN';
export const LOGOUT = 'app/LOGOUT';
export const FIRST_TIME = 'app/FIRST_TIME';
export const STAGES = 'app/STAGES';
export const SELECT_STAGE = 'app/SELECT_STAGE';
export const FIRST_NAME_CHANGED = 'app/FIRST_NAME_CHANGED';
export const LAST_NAME_CHANGED = 'app/LAST_NAME_CHANGED';
export const SAVE_NOTES = 'app/SAVE_NOTES';
export const PERSON_FIRST_NAME_CHANGED = 'app/PERSON_FIRST_NAME_CHANGED';
export const PERSON_LAST_NAME_CHANGED = 'app/PERSON_LAST_NAME_CHANGED';
export const ADD_STEP_REMINDER = 'app/ADD_STEP_REMINDER';
export const REMOVE_STEP_REMINDER = 'app/REMOVE_STEP_REMINDER';
export const PUSH_NOTIFICATION_SHOULD_ASK = 'app/PUSH_NOTIFICATION_SHOULD_ASK';
export const PUSH_NOTIFICATION_ASKED = 'app/PUSH_NOTIFICATION_ASKED';
export const PUSH_NOTIFICATION_SET_TOKEN = 'app/PUSH_NOTIFICATION_SET_TOKEN';
export const PUSH_NOTIFICATION_REMINDER = 'app/PUSH_NOTIFICATION_REMINDER';
export const PEOPLE_WITH_ORG_SECTIONS = 'app/PEOPLE_WITH_ORG_SECTIONS';
export const SWIPE_REMINDER_STEPS_HOME = 'app/SWIPE_REMINDER_STEPS_HOME';
export const SWIPE_REMINDER_STEPS_CONTACT = 'app/SWIPE_REMINDER_STEPS_CONTACT';
export const SWIPE_REMINDER_STEPS_REMINDER = 'app/SWIPE_REMINDER_STEPS_REMINDER';
export const SWIPE_REMINDER_JOURNEY = 'app/SWIPE_REMINDER_JOURNEY';
export const SET_VISIBLE_PERSON_INFO = 'app/SET_VISIBLE_PERSON_INFO';
export const UPDATE_VISIBLE_PERSON_INFO = 'app/UPDATE_VISIBLE_PERSON_INFO';
export const SET_JOURNEY_ITEMS = 'app/SET_JOURNEY_ITEMS';
export const CLEAR_JOURNEY_ITEMS = 'app/CLEAR_JOURNEY_ITEMS';
export const COMPLETED_STEP_COUNT = 'app/COMPLETED_STEP_COUNT';
export const UPDATE_STAGES = 'app/UPDATE_STAGES';
export const UPDATE_ONBOARDING_PERSON = 'app/UPDATE_ONBOARDING_PERSON';
export const RESET_ONBOARDING_PERSON = 'app/RESET_ONBOARDING_PERSON';

export const STEP_NOTE = 'stepNote';

export const NAVIGATE_FORWARD = 'Navigation/NAVIGATE';
export const NAVIGATE_RESET = 'Navigation/RESET';
export const MAIN_TABS = 'nav/MAIN_TABS';
export const DRAWER_OPEN = 'DrawerOpen';
export const DRAWER_CLOSE = 'DrawerClose';
export const MAIN_MENU_DRAWER = 'nav/drawer/main';
export const CONTACT_MENU_DRAWER = 'nav/drawer/contact';

export const EXPIRED_ACCESS_TOKEN = 'Expired access token';

export const URL_ENCODED = 'application/x-www-form-urlencoded';
export const THE_KEY_CLIENT_ID = '8480288430352167964';

export const CASEY = 'casey';
export const JEAN = 'jean';

export const LINKS = {
  about: 'https://get.missionhub.com',
  help: 'http://help.missionhub.com',
  playStore: 'market://details?id=com.missionhub',
  appleStore: 'itms://itunes.apple.com/us/app/apple-store/id447869440?mt=8',
  terms: 'https://get.missionhub.com/terms-of-service/',
};

export const ANALYTICS_CONTEXT_CHANGED = 'app/ANALYTICS_CONTEXT_CHANGED';

export const ORG_PERMISSIONS = [ 1, 4 ];

export const ANALYTICS = {
  PAGE_NAME: 'PageName',
  MCID: 'cru.mcid',
  SCREENNAME: 'cru.screenname',
  PREVIOUS_SCREENNAME: 'cru.previousscreenname',
  SITE_SECTION: 'cru.sitesection',
  SITE_SUBSECTION: 'cru.sitesubsection',
  SITE_SUB_SECTION_3: 'cru.subsectionlevel3',
  CONTENT_AUDIENCE_TARGET: 'cru.contentaudiencetarget',
  CONTENT_TOPIC: 'cru.contenttopic',
  LOGGED_IN_STATUS: 'cru.loggedinstatus',
  SSO_GUID: 'cru.ssoguid',
  GR_MASTER_PERSON_ID: 'cru.grmasterpersonid',
  FACEBOOK_ID: 'cru.facebookid',
  CONTENT_LANGUAGE: 'cru.contentlanguage',
};
export const LOGGED_IN = 'logged in';
export const NOT_LOGGED_IN = 'not logged in';

export const INTERACTION_TYPES = {
  MHInteractionTypeAssignedContacts: { id: 100, requestFieldName: 'contact_count', iconName: 'peopleIcon', translationKey: 'interactionAssignedContacts' },
  MHInteractionTypeUncontacted: { id: 101, requestFieldName: 'uncontacted_count', iconName: 'spiritualConversationsIcon', translationKey: 'interactionUncontacted' },
  MHInteractionTypeNote: { id: 1, iconName: 'commentIcon', translationKey: 'interactionNote', hideReport: true },
  MHInteractionTypeSpiritualConversation: { id: 2, iconName: 'spiritualConversationsIcon', translationKey: 'interactionSpiritualConversation' },
  MHInteractionTypeGospelPresentation: { id: 3, iconName: 'gospelIcon', translationKey: 'interactionGospel' },
  MHInteractionTypePersonalDecision: { id: 4, iconName: 'decisionIcon', translationKey: 'interactionDecision' },
  MHInteractionTypeHolySpiritConversation: { id: 5, iconName: 'spiritIcon', translationKey: 'interactionSpirit' },
  // MHInteractionTypeGraduatingOnMission: { id: 6, iconName: '', translationKey: 'interactionNote' },
  MHInteractionTypeDiscipleshipConversation: { id: 9, iconName: 'discipleshipConversationIcon', translationKey: 'interactionDiscipleshipConversation' },
};

export default {

};
