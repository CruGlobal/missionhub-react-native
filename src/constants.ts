/* eslint-disable max-lines */

import Config from 'react-native-config';

import { FeedItemSubjectTypeEnum } from '../__generated__/globalTypes';

export const LOGOUT = 'app/LOGOUT';
export type LogoutAction = { type: typeof LOGOUT };
export const PEOPLE_WITH_ORG_SECTIONS = 'app/PEOPLE_WITH_ORG_SECTIONS';
export const GROUP_INVITE_INFO = 'app/GROUP_INVITE_INFO';
export const GROUP_ONBOARDING_CARD = 'app/GROUP_ONBOARDING_CARD';
export const GROUP_TAB_SCROLL_ON_MOUNT = 'app/GROUP_TAB_SCROLL_ON_MOUNT';
export const LOAD_PERSON_DETAILS = 'app/LOAD_PERSON_DETAILS';
export const LOAD_ORGANIZATIONS = 'app/LOAD_ORGANIZATIONS';
export const UPDATE_PERSON_ATTRIBUTES = 'app/UPDATE_PERSON_ATTRIBUTES';
export const DELETE_PERSON = 'app/DELETE_PERSON';
export const REMOVE_ORGANIZATION_MEMBER = 'app/REMOVE_ORGANIZATION_MEMBER';
export const UPDATE_JOURNEY_ITEMS = 'app/UPDATE_JOURNEY_ITEMS';
export const GET_ORGANIZATIONS_CONTACTS_REPORT =
  'app/GET_ORGANIZATIONS_CONTACTS_REPORT';
export const RESET_STEP_COUNT = 'app/RESET_STEP_COUNT';
export const UPDATE_STAGES = 'app/UPDATE_STAGES';
export const UPDATE_CHALLENGE = 'app/UPDATE_CHALLENGE';
export const GET_ORGANIZATION_PEOPLE = 'app/GET_ORGANIZATION_PEOPLE';
export const RESET_CELEBRATION_PAGINATION = 'app/RESET_CELEBRATION_PAGINATION';
export const RESET_CHALLENGE_PAGINATION = 'app/RESET_CHALLENGE_PAGINATION';
export const RELOAD_APP = 'app/RELOAD_APP';

export const GLOBAL_COMMUNITY_ID = '_global_community_id';

export const EDIT_JOURNEY_STEP = 'editJourneyStep';
export const EDIT_JOURNEY_ITEM = 'editJourneyItem';
export const STEP_NOTE = 'stepNote';
export const CREATE_STEP = 'createStep';

export const ACCEPTED_STEP = 'accepted_challenge';

export const STEPS_TAB = 'StepsTab';
export const PEOPLE_TAB = 'PeopleTab';
export const COMMUNITIES_TAB = 'CommunitiesTab';
export const NOTIFICATIONS_TAB = 'NotificationsTab';

export const MAIN_TABS = 'nav/MAIN_TABS';
export const MAIN_MENU_DRAWER = 'nav/drawer/main';

export const SAVE_PENDING_POST = 'app/SAVE_PENDING_POST';
export const DELETE_PENDING_POST = 'app/DELETE_PENDING_POST';
export const PENDING_POST_FAILED = 'app/PENDING_POST_FAILED';
export const PENDING_POST_RETRY = 'app/PENDING_POST_RETRY';

// Errors
export const EXPIRED_ACCESS_TOKEN = 'Expired access token';
export const INVALID_ACCESS_TOKEN = 'Invalid access token';
export const INVALID_GRANT = 'invalid_grant';
export const NETWORK_REQUEST_FAILED = 'Network request failed';
export const CANNOT_EDIT_FIRST_NAME =
  'You are not allowed to edit first names of other MissionHub users';
export const ERROR_PERSON_PART_OF_ORG =
  'this person already has permissions on this organization';

export const URL_ENCODED = 'application/x-www-form-urlencoded';
export const URL_FORM_DATA = 'multipart/form-data';
export const THE_KEY_CLIENT_ID = Config.THE_KEY_CLIENT_ID;

export const LINKS = {
  blog: 'https://get.missionhub.com/blog/',
  about: 'https://get.missionhub.com',
  help: 'http://help.missionhub.com',
  shareStory: 'mailto:stories@missionhub.com',
  suggestStep: 'https://get.missionhub.com/suggest-a-step-of-faith/',
  playStore: 'market://details?id=com.missionhub',
  appleStore: 'itms://itunes.apple.com/us/app/apple-store/id447869440?mt=8',
  terms: 'https://get.missionhub.com/terms-of-service/',
  privacy: 'https://get.missionhub.com/privacy',
};

export const ORG_PERMISSIONS = {
  ADMIN: '1',
  USER: '4',
  CONTACT: '2',
  OWNER: '3',
};
export const GCM_SENDER_ID = Config.GCM_SENDER_ID;

//parameter names for the data we send to Analytics
export const ANALYTICS_MCID = 'cru.mcid';
export const ANALYTICS_SCREEN_NAME = 'cru.screenname';
export const ANALYTICS_SITE_SECTION = 'cru.sitesection';
export const ANALYTICS_SITE_SUBSECTION = 'cru.sitesubsection';
export const ANALYTICS_SITE_SUBSECTION_3 = 'cru.subsectionlevel3';
export const ANALYTICS_PREVIOUS_SCREEN_NAME = 'cru.previousscreenname';
export const ANALYTICS_APP_NAME = 'cru.appname';
export const ANALYTICS_LOGGED_IN_STATUS = 'cru.loggedinstatus';
export const ANALYTICS_SSO_GUID = 'cru.ssoguid';
export const ANALYTICS_GR_MASTER_PERSON_ID = 'cru.grmasterpersonid';
export const ANALYTICS_FACEBOOK_ID = 'cru.facebookid';
export const ANALYTICS_CONTENT_LANGUAGE = 'cru.contentlanguage';
export const ANALYTICS_SECTION_TYPE = 'cru.section-type';
export const ANALYTICS_ASSIGNMENT_TYPE = 'cru.assignment-type';
export const ANALYTICS_EDIT_MODE = 'cru.edit-mode';
export const ANALYTICS_PERMISSION_TYPE = 'cru.permission-type';

export const LOGGED_IN = 'logged in';
export const NOT_LOGGED_IN = 'not logged in';

export const ACTIONS = {
  PERSON_ADDED: {
    name: 'Person Added',
    key: 'cru.personadded',
  },
  STATUS_CHANGED: {
    name: 'Contact Status Changed',
    key: 'cru.contactstatuschanged',
  },
  EMAIL_ENGAGED: {
    name: 'Contact Engaged by Email',
    key: 'cru.emailiconengaged',
  },
  CALL_ENGAGED: {
    name: 'Contact Engaged by Phone',
    key: 'cru.calliconengaged',
  },
  TEXT_ENGAGED: {
    name: 'Contact Engaged by Text',
    key: 'cru.texticonengaged',
  },
  FILTER_ENGAGED: {
    name: 'Search Filter Engaged',
    key: 'cru.searchfilterengaged',
  },
  SEARCH_FILTER: {
    key: 'cru.searchfilter',
  },
  USER_ERROR: {
    name: 'User Signin Error',
    key: 'cru.usersigninerror',
  },
  SYSTEM_ERROR: {
    name: 'System Signin Error',
    key: 'cru.systemsigninerror',
  },
  ONBOARDING_STARTED: {
    name: 'Onboarding Started',
    key: 'cru.onboardingstarted',
  },
  ONBOARDING_COMPLETE: {
    name: 'Onboarding Complete',
    key: 'cru.onboardingcomplete',
  },
  SELF_STAGE_SELECTED: {
    name: 'Self Stage Selected',
    key: 'cru.selfselectedstage',
  },
  PERSON_STAGE_SELECTED: {
    name: 'Person Stage Selected',
    key: 'cru.personselectedstage',
  },
  STAGE_SELECTED: {
    key: 'cru.stageselected',
  },
  STEP_DETAIL: {
    name: 'Step of Faith Detail',
    key: 'cru.stepoffaithdetail',
  },
  STEPS_ADDED: {
    name: 'Step of Faith Added',
    key: 'cru.stepoffaithadded',
  },
  STEP_CREATED: {
    name: 'Step of Faith Created',
    key: 'cru.stepoffaithcreated',
  },
  POST_STEP_ADDED: {
    name: 'Add to My Steps',
    key: 'cru.addtomysteps',
  },
  STEP_POST_TYPE: {
    key: 'cru.stepposttype',
  },
  ALLOW: {
    name: 'Notification Permissions',
    key: 'cru.notificationsallowed',
  },
  NOT_NOW: {
    name: 'Notification Permissions',
    key: 'cru.notificationsnotallowed',
  },
  NO_REMINDERS: {
    name: 'Notification Reminders',
    key: 'cru.notoreminders',
  },
  STEP_PRIORITIZED: {
    name: 'Step Prioritized',
    key: 'cru.stepprioritized',
  },
  STEP_DEPRIORITIZED: {
    name: 'Step Deprioritized',
    key: 'cru.stepdeprioritized',
  },
  STEP_REMOVED: {
    name: 'Step Removed',
    key: 'cru.stepremoved',
  },
  STEP_COMPLETED: {
    name: 'Step Completed',
    key: 'cru.stepcompleted',
  },
  JOURNEY_EDITED: {
    name: 'Edit on Person Journey',
    key: 'cru.journeyedit',
  },
  INTERACTION: {
    name: 'Action Taken on Person',
    COMMENT: 'cru.commentadded',
    SOMETHING_COOL_HAPPENED: 'cru.somethingcoolhappenedadded',
    SPIRITUAL_CONVERSATION: 'cru.initiatinggospelconversations',
    GOSPEL_PRESENTATION: 'cru.presentingthegospel',
    PERSONAL_DECISION: 'cru.newprofessingbelievers',
    HOLY_SPIRIT_PRESENTATION: 'cru.presentingtheholyspirit',
    DISCIPLESHIP: 'cru.discipleshipconversation',
  },
  ITEM_LIKED: {
    name: 'Celebrate Item Liked',
    key: 'cru.celebrateitemliked',
  },
  CELEBRATE_COMMENT_ADDED: {
    name: 'Celebrate Comment Added',
    key: 'cru.sendcelebratecomment',
  },
  CELEBRATE_COMMENT_EDITED: {
    name: 'Celebrate Comment Edited',
    key: 'cru.editcelebratecomment',
  },
  CELEBRATE_COMMENT_DELETED: {
    name: 'Celebrate Comment Deleted',
    key: 'cru.deletecelebratecomment',
  },
  CELEBRATE_COMMENT_REPORTED: {
    name: 'Celebrate Comment Reported',
    key: 'cru.reportcelebratecomment',
  },
  ASSIGNED_TO_ME: {
    name: 'Assigned to Me',
    key: 'cru.assignedtome',
  },
  SELECT_COMMUNITY: {
    name: 'Community Selected',
    key: 'cru.selectcommunities',
  },
  SELECT_CREATED_COMMUNITY: {
    name: 'Community Selected After Create',
    key: 'cru.createcommunities',
  },
  SELECT_JOINED_COMMUNITY: {
    name: 'Community Selected After Join',
    key: 'cru.joincommunities',
  },
  SEARCH_COMMUNITY_WITH_CODE: {
    name: 'Search for Community With Code',
    key: 'cru.codesearch',
  },
  JOIN_COMMUNITY_WITH_CODE: {
    name: 'Join Community With Code',
    key: 'cru.codejoin',
  },
  ADD_COMMUNITY_PHOTO: {
    name: 'Add Photo for Community',
    key: 'cru.communityphoto',
  },
  CREATE_COMMUNITY: {
    name: 'Create Community',
    key: 'cru.communitycreate',
  },
  CHALLENGE_CREATED: {
    name: 'Challenge Created',
    key: 'cru.challengecreated',
  },
  CHALLENGE_JOINED: {
    name: 'Challenge Joined',
    key: 'cru.challengejoined',
  },
  CHALLENGE_COMPLETED: {
    name: 'Challenge Completed',
    key: 'cru.challengecompleted',
  },
  CHALLENGE_DETAIL: {
    name: 'Challenge Detail View',
    key: 'cru.challengedetail',
  },
  MANAGE_MAKE_ADMIN: {
    name: 'Make Admin',
    key: 'cru.membersmakeadmin',
  },
  MANAGE_MAKE_OWNER: {
    name: 'Make Owner',
    key: 'cru.membersmakeowner',
  },
  MANAGE_REMOVE_ADMIN: {
    name: 'Remove Admin',
    key: 'cru.membersremoveadmin',
  },
  MANAGE_REMOVE_MEMBER: {
    name: 'Remove Member',
    key: 'cru.membersremovemember',
  },
  MANAGE_LEAVE_COMMUNITY: {
    name: 'Leave Community',
    key: 'cru.membersleave',
  },
  COMMUNITY_EDIT: {
    name: 'Editting Community',
    key: 'cru.communityedit',
  },
  COPY_CODE: {
    name: 'Copy Community Code',
    key: 'cru.copycode',
  },
  COPY_INVITE_URL: {
    name: 'Copy Community Invite URL',
    key: 'cru.copyinviteurl',
  },
  SEND_COMMUNITY_INVITE: {
    name: 'Send Community Invite',
    key: 'cru.sendcommunityinvite',
  },
  NEW_CODE: {
    name: 'New Community Code',
    key: 'cru.newcode',
  },
  NEW_INVITE_URL: {
    name: 'New Community Invite URL',
    key: 'cru.newinviteurl',
  },
  COMMUNITY_DELETE: {
    name: 'Delete Community',
    key: 'cru.communitydelete',
  },
  CREATE_POST: {
    name: 'Create Post',
    key: 'cru.posttype',
  },
  PHOTO_ADDED: {
    name: 'Add Photo to Post',
    key: 'cru.addphototopost',
  },
  VIDEO_ADDED: {
    name: 'Add Video to Post',
    key: 'cru.addvideotopost',
  },
  POST_TYPE_SELECTED: {
    name: 'Post Type Selected',
    key: 'cru.posttype',
  },
};

export const CONTACT_STEPS = 'Contact Steps';

export const INTERACTION_TYPES = {
  MHInteractionTypeAssignedContacts: {
    id: '100',
    requestFieldName: 'contact_count',
    iconName: 'peopleIcon',
    translationKey: 'interactionAssignedContacts',
  },
  MHInteractionTypeUncontacted: {
    id: '101',
    requestFieldName: 'uncontacted_count',
    iconName: 'uncontactedIcon',
    translationKey: 'interactionUncontacted',
  },
  MHInteractionTypeNote: {
    id: '1',
    iconName: 'commentIcon',
    translationKey: 'interactionNote',
    hideReport: true,
    isOnAction: true,
    tracking: ACTIONS.INTERACTION.COMMENT,
  },
  MHInteractionTypeSomethingCoolHappened: {
    id: '11',
    iconName: 'celebrateIcon',
    translationKey: 'interactionSomethingCoolHappened',
    hideReport: true,
    tracking: ACTIONS.INTERACTION.SOMETHING_COOL_HAPPENED,
  },
  MHInteractionTypeSpiritualConversation: {
    id: '2',
    iconName: 'spiritualConversationIcon',
    translationKey: 'interactionSpiritualConversation',
    isOnAction: true,
    tracking: ACTIONS.INTERACTION.SPIRITUAL_CONVERSATION,
  },
  MHInteractionTypeGospelPresentation: {
    id: '3',
    iconName: 'gospelIcon',
    translationKey: 'interactionGospel',
    isOnAction: true,
    tracking: ACTIONS.INTERACTION.GOSPEL_PRESENTATION,
  },
  MHInteractionTypePersonalDecision: {
    id: '4',
    iconName: 'decisionIcon',
    translationKey: 'interactionDecision',
    isOnAction: true,
    tracking: ACTIONS.INTERACTION.PERSONAL_DECISION,
  },
  MHInteractionTypeHolySpiritConversation: {
    id: '5',
    iconName: 'spiritIcon',
    translationKey: 'interactionSpirit',
    isOnAction: true,
    tracking: ACTIONS.INTERACTION.HOLY_SPIRIT_PRESENTATION,
  },
  MHInteractionTypeDiscipleshipConversation: {
    id: '9',
    iconName: 'discipleshipConversationIcon',
    translationKey: 'interactionDiscipleshipConversation',
    isOnAction: true,
    tracking: ACTIONS.INTERACTION.DISCIPLESHIP,
  },
};

export const DEFAULT_PAGE_LIMIT = 25;

export const DAYS_OF_THE_WEEK = [
  'sunday',
  'monday',
  'tuesday',
  'wednesday',
  'thursday',
  'friday',
  'saturday',
];

export enum NOTIFICATION_PROMPT_TYPES {
  ONBOARDING,
  FOCUS_STEP,
  LOGIN,
  SET_REMINDER,
  JOIN_COMMUNITY,
  JOIN_CHALLENGE,
}

export const DEFAULT_SUBJECT_TYPES = [
  FeedItemSubjectTypeEnum.STORY,
  FeedItemSubjectTypeEnum.QUESTION,
  FeedItemSubjectTypeEnum.PRAYER_REQUEST,
  FeedItemSubjectTypeEnum.ANNOUNCEMENT,
  FeedItemSubjectTypeEnum.HELP_REQUEST,
  FeedItemSubjectTypeEnum.THOUGHT,
  FeedItemSubjectTypeEnum.STEP,
  FeedItemSubjectTypeEnum.ACCEPTED_COMMUNITY_CHALLENGE,
  FeedItemSubjectTypeEnum.COMMUNITY_PERMISSION,
];
