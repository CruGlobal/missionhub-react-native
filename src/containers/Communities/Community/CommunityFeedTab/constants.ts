export const COMMUNITY_FEED = 'nav/COMMUNITY_FEED';
