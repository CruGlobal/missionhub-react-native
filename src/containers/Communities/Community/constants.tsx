import React from 'react';
import i18next from 'i18next';

import GroupChallenges, {
  COMMUNITY_CHALLENGES,
} from '../../Groups/GroupChallenges';

import {
  CommunityFeedTab,
  COMMUNITY_FEED,
} from './CommunityFeedTab/CommunityFeedTab';
import {
  COMMUNITY_IMPACT,
  CommunityImpactTab,
} from './CommunityImpactTab/CommunityImpactTab';
import { CommunitiesCollapsibleHeaderContext } from './CommunityHeader/CommunityHeader';

// Moved here to prevent circular dependency
export const COMMUNITY_TABS = 'nav/COMMUNITY_TABS';

export const communityTabs = [
  {
    name: i18next.t('communityTabs:feed'),
    navigationAction: COMMUNITY_FEED,
    component: () => (
      <CommunityFeedTab
        collapsibleHeaderContext={CommunitiesCollapsibleHeaderContext}
      />
    ),
  },
  {
    name: i18next.t('communityTabs:challenges'),
    navigationAction: COMMUNITY_CHALLENGES,
    component: GroupChallenges,
  },
  {
    name: i18next.t('communityTabs:impact'),
    navigationAction: COMMUNITY_IMPACT,
    component: () => (
      <CommunityImpactTab
        collapsibleHeaderContext={CommunitiesCollapsibleHeaderContext}
      />
    ),
  },
];
