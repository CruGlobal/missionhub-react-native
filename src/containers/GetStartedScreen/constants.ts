export const GET_STARTED_SCREEN = 'nav/GET_STARTED';
