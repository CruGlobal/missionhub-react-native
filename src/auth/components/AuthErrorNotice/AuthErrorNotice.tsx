import React, { useCallback } from 'react';
import { Text } from 'react-native';
import { SafeAreaView } from 'react-navigation';
import { useTranslation } from 'react-i18next';

import { AuthError } from '../../constants';

import { styles } from './styles';

interface AuthErrorNoticeProps {
  error: AuthError;
}

export const AuthErrorNotice = ({ error }: AuthErrorNoticeProps) => {
  const { t } = useTranslation('authErrorNotice');

  const renderMessage = useCallback(() => {
    switch (error) {
      case AuthError.CredentialsIncorrect:
        return t('credentialsIncorrect');
      case AuthError.EmailUnverified:
        return t('emailUnverified');
      case AuthError.MfaIncorrect:
        return t('mfaIncorrect');
      case AuthError.Unknown:
      default:
        return t('unknown');
    }
  }, [error]);

  return error !== AuthError.None && error !== AuthError.MfaRequired ? (
    <SafeAreaView style={styles.errorBar}>
      <Text style={styles.errorMessage}>{renderMessage()}</Text>
    </SafeAreaView>
  ) : null;
};
