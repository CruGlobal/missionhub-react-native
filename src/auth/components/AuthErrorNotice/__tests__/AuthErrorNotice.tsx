import React from 'react';
import { render } from 'react-native-testing-library';
import i18next from 'i18next';

import { AuthErrorNotice } from '../AuthErrorNotice';
import { AuthError } from '../../../constants';

it('should match snapshot', () => {
  const { toJSON } = render(
    <AuthErrorNotice error={AuthError.CredentialsIncorrect} />,
  );

  expect(toJSON()).toMatchSnapshot();
});

it.each([
  ['credentialsIncorrect', AuthError.CredentialsIncorrect],
  ['emailUnverified', AuthError.EmailUnverified],
  ['mfaIncorrect', AuthError.MfaIncorrect],
  ['unknown', AuthError.Unknown],
])('should render %s', (i18nValue, authError) => {
  const { getByText } = render(<AuthErrorNotice error={authError} />);

  expect(() =>
    getByText(i18next.t(`authErrorNotice:${i18nValue}`)),
  ).not.toThrow();
});
