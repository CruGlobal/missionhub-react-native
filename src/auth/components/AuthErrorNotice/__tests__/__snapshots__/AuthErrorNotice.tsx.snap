// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<View
  forwardedRef={[Function]}
  onLayout={[Function]}
  pointerEvents="box-none"
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FF5532",
      "height": 68,
      "justifyContent": "center",
      "paddingBottom": 0,
      "paddingLeft": 0,
      "paddingRight": 0,
      "paddingTop": 20,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#ffffff",
        "fontFamily": "SourceSansPro-Regular",
        "fontSize": 16,
        "fontStyle": "normal",
        "lineHeight": 24,
        "textAlignVertical": "center",
        "textTransform": "none",
      }
    }
  >
    Your Email or Password is Incorrect
  </Text>
</View>
`;
